src/combinatorics/simple_graph/clique.lean

```diff
@@ -123,6 +137,57 @@ def clique_free (n : ℕ) : Prop := ∀ t, ¬ G.is_n_clique n t
 
 variables {G H}
 
+lemma not_clique_free_of_top_embedding {n : ℕ}
+  (f : (⊤ : simple_graph (fin n)) ↪g G) : ¬ G.clique_free n :=
+begin
+  simp only [clique_free, is_n_clique_iff, is_clique_iff_induce_eq, not_forall, not_not],
+  use finset.univ.map f.to_embedding,
+  simp only [card_map, finset.card_fin, eq_self_iff_true, and_true],
+  ext ⟨v, hv⟩ ⟨w, hw⟩,
+  simp only [coe_map, rel_embedding.coe_fn_to_embedding, set.mem_image,
+    coe_univ, set.mem_univ, true_and] at hv hw,
+  obtain ⟨v', rfl⟩ := hv,
+  obtain ⟨w', rfl⟩ := hw,
+  simp only [f.map_adj_iff, comap_adj, function.embedding.coe_subtype, subtype.coe_mk, top_adj,
+    ne.def, subtype.mk_eq_mk],
+  exact (function.embedding.apply_eq_iff_eq _ _ _).symm.not,
+end
+
+/-- An embedding of a complete graph that witnesses the fact that the graph is not clique-free. -/
+noncomputable
+def top_embedding_of_not_clique_free {n : ℕ} (h : ¬ G.clique_free n) :
+  (⊤ : simple_graph (fin n)) ↪g G :=
+begin
+  simp only [clique_free, is_n_clique_iff, is_clique_iff_induce_eq, not_forall, not_not] at h,
+  obtain ⟨ha, hb⟩ := h.some_spec,
+  have : (⊤ : simple_graph (fin h.some.card)) ≃g (⊤ : simple_graph h.some),
+  { apply iso.complete_graph,
+    simpa using (fintype.equiv_fin h.some).symm },
+  rw ← ha at this,
+  convert (embedding.induce ↑h.some).comp this.to_embedding;
+  exact hb.symm,
+end
+
+lemma not_clique_free_iff (n : ℕ) :
+  ¬ G.clique_free n ↔ nonempty ((⊤ : simple_graph (fin n)) ↪g G) :=
+begin
+  split,
+  { exact λ h, ⟨top_embedding_of_not_clique_free h⟩ },
+  { rintro ⟨f⟩,
+    exact not_clique_free_of_top_embedding f },
+end
+
+lemma clique_free_iff {n : ℕ} :
+  G.clique_free n ↔ is_empty ((⊤ : simple_graph (fin n)) ↪g G) :=
```

Comment:
Not sure about this one, but maybe turn it around and simp it? I guess it depends on whether we will ever use `is_empty (G ↪g H)` commonly as an hypothesis.

Reply:
(I have the same concern here as with `is_clique_iff_induce_eq `.)